File: LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/UntrustedDomainPatch.java
package com.aliucord.plugins.layoutcontroller.patchers;

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher;
import com.aliucord.plugins.layoutcontroller.util.Const;
import com.discord.stores.StoreMaskedLinks;

import top.canyie.pine.Pine;

public class UntrustedDomainPatch extends BasePatcher {

    public UntrustedDomainPatch() throws Exception {
        super(Const.Key.UNTRUSTED_DOMAINS_KEY, Const.ViewName.UNTRUSTED_DOMAINS_NAME, StoreMaskedLinks.class.getDeclaredMethod("isTrustedDomain", String.class, String.class));
    }

    @Override
    public void patchBody(Pine.CallFrame callFrame) {
        callFrame.setResult(true);
    }
}

File: plugins/LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/UntrustedDomainPatch.kt
package com.aliucord.plugins.layoutcontroller.patchers

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher
import com.aliucord.plugins.layoutcontroller.util.Description
import com.aliucord.plugins.layoutcontroller.util.Key
import com.discord.stores.StoreMaskedLinks
import de.robv.android.xposed.XC_MethodHook

class UntrustedDomainPatch : BasePatcher(
    key = Key.UNTRUSTED_DOMAINS_KEY,
    description = Description.UNTRUSTED_DOMAINS_DESCRIPTION,
    classMember = StoreMaskedLinks::class.java.getDeclaredMethod(
        "isTrustedDomain",
        String::class.java,
        String::class.java
    )
) {
    override fun patchBody(callFrame: XC_MethodHook.MethodHookParam) {
        callFrame.result = true
    }
}